zulipterminal/ui_tools/boxes.py

```diff
@@ -480,19 +483,22 @@ def keypress(self, size: Tuple[int, int], key: str) -> str:
                 self.model.controller.narrow_to_topic(self)
         elif is_command_key('GO_BACK', key):
             self.model.controller.show_all_messages(self)
-        elif is_command_key('REPLY_AUTHOR', key):
+        elif is_command_key('REPLY_AUTHOR', key) and \
+                not self.is_in_empty_narrow:
             self.model.controller.view.write_box.private_box_view(
                 email=self.message['sender_email']
             )
-        elif is_command_key('MENTION_REPLY', key):
+        elif is_command_key('MENTION_REPLY', key) and \
+                not self.is_in_empty_narrow:
             self.keypress(size, 'enter')
             mention = '@**' + self.message['sender_full_name'] + '** '
             self.model.controller.view.write_box.msg_write_box.set_edit_text(
                 mention)
             self.model.controller.view.write_box.msg_write_box.set_edit_pos(
                 len(mention))
             self.model.controller.view.middle_column.set_focus('footer')
-        elif is_command_key('QUOTE_REPLY', key):
+        elif is_command_key('QUOTE_REPLY', key) and \
+                not self.is_in_empty_narrow:
```

Comment:
So a map to what should and should-not be allowed:

Hot keys for dummy_message on PM: 
* r - Y
* R - N (sender is WelcomeBot and not recipient)
* @ - N
* > - N
* c - Y(Same effect as r)
* x - Y
* thumbs up - N
* star status - N
* z/P - Y (works normally)
------
Hot keys for dummy_message on Stream:
* r - Should work the same as c - with no topic filled.
      (To be done).

Reply:
This was discussed on `czo`.

Reply:
@neiljp The thumb-up and star reactions are automatically taken care. It is because the dummy-message is not really in Focus. This is due to addition d3248d7602a9a9c0062e9402b42575f5f0aac622, which checks if `self.focus is not None`.

Perhaps this can be taken care of in a better way. I am hesitant to add more code here as any changes need be done to be done to `views.py` (requiring significant additions atleast).